workflow/rules/common.smk

```diff
@@ -50,6 +50,6 @@ def get_sample_fastq(wildcards):
 
 def compile_output_list(wildcards):
     output_list = []
-    output_list += expand("alignment/bwa_mem/{sample}_{unit}.bam", sample=samples.index)
-    output_list += expand("alignment/bwa_mem/{sample}_{unit}.bam.bai", sample=samples.index)
+    output_list += expand("alignment/bwa_mem/{sample}_{unit}.bam", sample=samples.index, unit=units.unit)
```

Comment:
I am not sure if this works. I think what `expand` does, it takes both lists of items and puts out all possible combinations which is not exactly what the target should be. for each `sample` only the available `unit` should be used.